pulsar-client/src/main/java/org/apache/pulsar/client/impl/auth/oauth2/AuthenticationDataOAuth2.java

```diff
@@ -44,7 +42,10 @@ public boolean hasDataForHttp() {
 
     @Override
     public Set<Map.Entry<String, String>> getHttpHeaders() {
-        return this.headers;
```

Comment:
What is the purpose of replacing the stored headers map with a computed map? The `accessToken` is `final` in this class, so I wouldn't think we need to compute the map each time.